### PR TITLE
Fix write_gnuplot with Eulerian mappings.

### PR DESCRIPTION
**Pull request by drwells**
Follow up to #6632 that maps vertices correctly.

### CODE REVIEW

File: source/grid/grid_out.cc [hunk 1]
```diff
@@ -3303,11 +3303,9 @@ namespace internal
                       const unsigned int offset=face_no*n_points;
                       // we don't need to transform the vertices: they are
                       // already in physical space.
-                      line_points.push_back(face->vertex(0));
-                      for (unsigned int i=1; i<n_points - 1; ++i)
+                      for (unsigned int i=0; i<n_points; ++i)
```
Comment (bangerth) [inline source/grid/grid_out.cc]: Ha, neat. That makes the code even nicer.

But you need to update the comment above. Probably same in the duplicated code below.

**Comment by drwells** [comment]
and this is already out of date :laughing:

**Comment by bangerth** [comment]
You were too slow in fixing the wrong comment.....

**Comment by drwells** [comment]
Actually, while looking over this again, I realized that there are more places where we assume that the mapping does not move vertices. I am working on this now and will include a few more fixes.

**Comment by bangerth** [comment]
Are they related to #6632? If not, keep it separate and let's just get this one thing merged here.

**Comment by drwells** [comment]
fair enough. I fixed the outstanding problems.

**Comment by bangerth** [comment]
Thanks!

**Pull request merged**